cadc-sia/src/main/java/ca/nrc/cadc/sia2/impl/ServiceAvailability.java

```diff
@@ -130,27 +146,60 @@ public void setState(String string)
     {
         //no-op
     }
-    
+
+    /**
+     * Obtain the base TAP URL to use.  This method will read in an optionally configured 'tapURI' property and if it
+     * appears to be a URL, then assume an unregistered TAP service was configured and treat it as the base URL,
+     * otherwise attempt to use it as a URI and look the URL up in the Registry.
+     *
+     * @return  URL     Base URL of the TAP service to use.
+     * @throws Exception    If a URL cannot be created from the specified string.
+     */
+    public static URL getTapBaseURL() throws Exception
+    {
+        RegistryClient regClient = new RegistryClient();
+        URI configuredTapURI = URI.create(ServiceAvailability.getTapURI());
+
+        if (configuredTapURI.getScheme().startsWith("http"))
+        {
+            return configuredTapURI.toURL();
+        }
+        else
+        {
+            AuthMethod am = AuthenticationUtil.getAuthMethod(AuthenticationUtil.getCurrentSubject());
+            return regClient.getServiceURL(configuredTapURI, Standards.TAP_10, (am == null) ? AuthMethod.ANON : am);
+        }
+    }
```

Comment:
I wonder if the order of operations should be switched here... first check if the reg client returns a URL.  If not, just return uri.toURL().  That would avoid the 'if http' check.
You'd have to catch the MalformedURLException on the toURL() call and return null.

Reply:
Done.

Reply:
try/catch around configuredTapURI.toURL() to catch MalformedURLException so that null can be returned...

Reply:
Just one more comment Dustin:  Could you change the signature of this method so that it throws MalformedURLException (and potentially others) rather than blanket Exception?